website/content/docs/intro/use-cases/dynamic-config-vault-dynamic-secrets.mdx

```diff
@@ -0,0 +1,269 @@
+---
+layout: docs
+page_title: Dynamic App Config using Vault Dynamic Secrets
+description: |-
+  Waypoint enables you to source config dynamically from various external data sources using config sourcer plugins - this use case depicts doing just that with Vault dynamic database secrets.
+---
+
+# Dynamic Secrets for Waypoint with Vault
+
+Hashicorp Waypoint is an application deployment platform that simplifies how users
+test and release apps onto platforms and utilize infrastructure managed by Terraform.
+After deploying your app, Waypoint can also be used to configure it! This can be
+done with static application configuration, where static environment variables
+or files can be applied to your running deployment. Alternatively, configuration
+can be sourced dynamically from external sources, using a config sourcer plugin.
+
+The [Hashicorp Vault plugin for Waypoint](https://developer.hashicorp.com/waypoint/plugins/vault)
+is one of the [config sourcer plugins](https://developer.hashicorp.com/waypoint/docs/app-config/dynamic)
+that is built-in to the Waypoint binary. This plugin can be used to source static
+values from the Vault KV secrets engine, but also for dynamic secrets! A [dynamic secret](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-dynamic-secrets)
+is one that is managed by Vault and has a lease associated with it, where
+the secret will expire after a set time to live (TTL). This is a huge benefit to
+the security of your application, as any possibly leaked secrets would automatically
+expire after a pre-configured amount of time, among other things including that
+developers and operators don’t need to know secrets for their deployed applications.
+
+Prior to using Waypoint to deploy my Nomad jobs, I’ve relied on [Nomad’s integration
+with Vault](https://developer.hashicorp.com/nomad/docs/integrations/vault-integration)
+to template secrets to my applications. I still do make use of this integration,
+but in instances where I use Waypoint to deploy my app, I can abstract away that
+integration by using Waypoint’s dynamic configuration integration with Vault, even
+when I’m not deploying to my Nomad cluster. Let’s look into how I configured this
+in Waypoint.
+
+My application, located at [this path](https://github.com/hashicorp/waypoint-examples/tree/dynamic-config-example/learn/dynamic-application-configuration)
```

Comment:
For safety sake, let's point this at a specific commit rather than `main` in case the link goes stale

Reply:
I agree - left this on my separate branch since this change isn't merged to `main` yet, but I'll open a PR on the examples repo for that pretty soon!